src/commands.cpp

```diff
@@ -1998,173 +1998,281 @@ void CommandUserRequest::procresult()
-                                                 crypto_sign_PUBLICKEYBYTES));
-            }
+            client->initializekeys(); // we have now all the required data
         }
 #endif
-
-        return(client->app->getua_result(e));
+        return;
     }
     else
     {
         const char* ptr;
         const char* end;
-
-        if (!(ptr = client->json.getvalue()) || !(end = strchr(ptr, '"')))
+        string v;
+        for (;;)
         {
-            return(client->app->getua_result(API_EINTERNAL));
-        }
+            switch (client->json.getnameid())
+            {
+                case MAKENAMEID2('a','v'):
+                    if (!(ptr = client->json.getvalue()) || !(end = strchr(ptr, '"')))
+                    {
+                        client->app->getua_result(API_EINTERNAL);
+                        return;
+                    }
+                    break;
 
-        // if there's no avatar, the value is "none" (not Base64 encoded)
-        if (attributename == "+a" && !strncmp(ptr, "none", 4))
-        {
-            return(client->app->getua_result(API_ENOENT));
-        }
+                case 'v':
+                {
+                    const char* vptr;
+                    const char* vend;
+                    if (!(vptr = client->json.getvalue()) || !(vend = strchr(vptr, '"')))
+                    {
+                        client->app->getua_result(API_EINTERNAL);
+                        return;
+                    }
+                    v.assign(vptr, (vend-vptr));
+                    break;
+                }
+                case EOO:
+                {
+                    // if there's no avatar, the value is "none" (not Base64 encoded)
+                    if (u && an == "+a" && !strncmp(ptr, "none", 4))
+                    {
+                        u->setattr(&an, NULL, &v);
+                        client->app->getua_result(API_ENOENT);
+                        client->notifyuser(u);
+                        return;
+                    }
 
-        int l;
-        byte* data;
-        l = (end - ptr) / 4 * 3 + 3;
-        data = new byte[l];
-        l = Base64::atob(ptr, data, l);
+                    // convert from ASCII to binary the received data
+                    string av;
+                    av.resize((end - ptr) / 4 * 3 + 3);
+                    av.resize(Base64::atob(ptr, (byte *)av.data(), av.size()));
 
-        if (attributename == "firstname")
-        {
-            if (!user->firstname)
-            {
-                user->firstname = new string;
-            }
-            user->firstname->assign((char*) data, l);
-        }
-        else if (attributename == "lastname")
-        {
-            if (!user->lastname)
-            {
-                user->lastname = new string;
-            }
-            user->lastname->assign((char*) data, l);
-        }
+                    // bool nonHistoric = (attributename.at(1) == '!');
```

Comment:
Yes, right now it's not going to be used. But I'd like to keep it there for reference and for a near future where SDK might need to adapt the behaviour based on the historical property of attributes.